src/steps/domains/index.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  createDirectRelationship,
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  RelationshipClass,
+} from '@jupiterone/integration-sdk-core';
+
+import { IntegrationConfig } from '../../config';
+import { createAPIClient } from '../../client';
+import { Entities, Relationships, IntegrationSteps } from '../constants';
+import { createDomainEntity } from './converters';
+import { getInstallKey } from '../installs/converters';
+
+export async function fetchDomains({
+  instance,
+  jobState,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const apiClient = createAPIClient(instance.config);
+
+  await jobState.iterateEntities(Entities.INSTALL, async (install) => {
+    await apiClient.iterateDomains(install.id as string, async (domain) => {
+      const domainEntity = createDomainEntity(domain);
+
+      await jobState.addEntity(domainEntity);
+
+      const installEntity = await jobState.findEntity(
+        getInstallKey(install.id as string),
+      );
+
+      if (installEntity && domainEntity) {
+        await jobState.addRelationship(
+          createDirectRelationship({
+            _class: RelationshipClass.HAS,
+            from: installEntity,
+            to: domainEntity,
+          }),
+        );
+      }
+    });
+  });
+}
```

Comment:
I've got 1 suggestion and 1 correction for this:

Suggestion: whenever you use `jobState.iterateEntities`, try naming the returned object as "nameEntity", so in this case it'll be: `async (install)` -> `async (installEntity)`.

Correction: the returned `install` object from `jobState.iterateEntities` is already of Entity type (suggestion above helps with this in the future :slightly_smiling_face: ) so you don't need to use `jobState.findEntity()` on L26